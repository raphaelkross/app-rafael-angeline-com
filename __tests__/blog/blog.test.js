/* eslint-env jest */

import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Blog from "../../components/blog/blog";
import Post from "../../components/blog/post";
import Categories from "../../components/blog/categories";
import Pagination from "../../components/blog/pagination";

import PostsMock from "../../lib/mocks/posts";
import CategoriesMock from "../../lib/mocks/categories";

describe("Blog", () => {
	it("should render proper number of <Post />", () => {
		const blog = shallow(
			<Blog
				posts={PostsMock}
				categories={CategoriesMock}
				current={1}
				pages={5}
				title="Blog Page"
			/>
		);

		expect(blog.find(Post).length).toEqual(4);
	});

	it("should render message when no post is rendered", () => {
		const blog = shallow(
			<Blog
				posts={[]}
				categories={CategoriesMock}
				current={1}
				pages={5}
				title="Blog Page"
			/>
		);

		expect(blog.find(".no-results").length).toEqual(1);
	});

	it("should render <Categories /> widget", () => {
		const blog = shallow(
			<Blog
				posts={PostsMock}
				categories={CategoriesMock}
				current={1}
				pages={5}
				title="Blog Page"
			/>
		);

		expect(blog.find(Categories).length).toEqual(1);
	});

	it("should be calling <pagination /> and with proper parameters", () => {
		const blog = shallow(
			<Blog
				posts={PostsMock}
				categories={CategoriesMock}
				current={1}
				pages={5}
				title="Blog Page"
			/>
		);

		expect(blog.find(Pagination).length).toEqual(1);
		expect(blog.find(Pagination).prop("current")).toEqual(1);
		expect(blog.find(Pagination).prop("pages")).toEqual(5);
	});

	it("should match complete layout snapshot", () => {
		const component = renderer.create(
			<Blog
				posts={PostsMock}
				categories={CategoriesMock}
				current={1}
				pages={5}
				title="Blog Page"
			/>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
