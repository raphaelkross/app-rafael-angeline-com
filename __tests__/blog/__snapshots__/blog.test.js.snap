// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog should match complete layout snapshot 1`] = `
Array [
  <header
    className="jsx-3590052481"
  >
    <div
      className="jsx-3590052481 container"
    >
      <nav
        className="jsx-3590052481"
      >
        <ul>
          <li>
            <a
              href="http://localhost:3000/"
              onClick={[Function]}
            >
              <span>
                Home.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/about"
              onClick={[Function]}
            >
              <span>
                About.
              </span>
            </a>
            <ul>
              <li>
                <a
                  href="https://twitter.com/RafaelAngeline"
                  target="_blank"
                >
                  <span>
                    Twitter.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/raphaelkross"
                  target="_blank"
                >
                  <span>
                    Github.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://profiles.wordpress.org/rafaelangeline"
                  target="_blank"
                >
                  <span>
                    WordPress.
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li>
            <a
              href="http://localhost:3000/work"
              onClick={[Function]}
            >
              <span>
                Work.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/contact"
              onClick={[Function]}
            >
              <span>
                Contact.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/blog"
              onClick={[Function]}
            >
              <span>
                Blog.
              </span>
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="jsx-3590052481 responsive-trigger"
      >
        <a
          className="jsx-3590052481 responsive-trigger-link"
          href="#"
          onClick={[Function]}
        >
          <svg
            className="jsx-3590052481 feather feather-menu"
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              className="jsx-3590052481"
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              className="jsx-3590052481"
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              className="jsx-3590052481"
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="jsx-3590052481 modal"
    >
      <a
        className="jsx-3590052481 modal-close"
        href="#"
        onClick={[Function]}
      >
        <svg
          className="jsx-3590052481 feather feather-x"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            className="jsx-3590052481"
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            className="jsx-3590052481"
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </a>
      <div
        className="jsx-3590052481"
      >
        <ul>
          <li>
            <a
              href="http://localhost:3000/"
              onClick={[Function]}
            >
              <span>
                Home.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/about"
              onClick={[Function]}
            >
              <span>
                About.
              </span>
            </a>
            <ul>
              <li>
                <a
                  href="https://twitter.com/RafaelAngeline"
                  target="_blank"
                >
                  <span>
                    Twitter.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/raphaelkross"
                  target="_blank"
                >
                  <span>
                    Github.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://profiles.wordpress.org/rafaelangeline"
                  target="_blank"
                >
                  <span>
                    WordPress.
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li>
            <a
              href="http://localhost:3000/work"
              onClick={[Function]}
            >
              <span>
                Work.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/contact"
              onClick={[Function]}
            >
              <span>
                Contact.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/blog"
              onClick={[Function]}
            >
              <span>
                Blog.
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </header>,
  <main
    className="jsx-1079858574"
  >
    <div
      className="jsx-1079858574 container"
    >
      <div
        className="jsx-1079858574 content"
      >
        <div
          className="jsx-1079858574 grid"
        >
          <div
            className="jsx-675934160 post"
          >
            <p
              className="jsx-675934160 date"
            >
              May 9, 2019
            </p>
            <h2
              className="jsx-675934160 title"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                Personal website overhaul.
              </a>
            </h2>
            <picture
              className="jsx-675934160"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                <img
                  alt="Alt Text"
                  className="jsx-675934160"
                  src="https://images.unsplash.com/photo-1508704019882-f9cf40e475b4?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=8c6e5e3aba713b17aa1fe71ab4f0ae5b&auto=format&fit=crop&w=800&q=60"
                  title="Image Title"
                />
              </a>
            </picture>
            <div
              className="jsx-675934160 excerpt"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "WordPress thumbnails is a really used feature, learn how to
		customize the image sizes from WordPress before activating a
		new theme.<br><br>
		Using the default sizes instead custom image sizes avoid
		longer media upload times and a smooth experience to your
		users.",
                }
              }
            />
            <p
              className="jsx-675934160 post-details"
            >
              Posted by 
              <span
                className="jsx-675934160"
              >
                @
                rafael angeline
              </span>
               
              in 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#cat"
                onClick={[Function]}
              >
                wordpress
              </a>
              , 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#dev"
                onClick={[Function]}
              >
                development
              </a>
            </p>
          </div>
          <div
            className="jsx-675934160 post"
          >
            <p
              className="jsx-675934160 date"
            >
              May 9, 2019
            </p>
            <h2
              className="jsx-675934160 title"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                Personal website overhaul.
              </a>
            </h2>
            <picture
              className="jsx-675934160"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                <img
                  alt="Alt Text"
                  className="jsx-675934160"
                  src="https://images.unsplash.com/photo-1508704019882-f9cf40e475b4?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=8c6e5e3aba713b17aa1fe71ab4f0ae5b&auto=format&fit=crop&w=800&q=60"
                  title="Image Title"
                />
              </a>
            </picture>
            <div
              className="jsx-675934160 excerpt"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "WordPress thumbnails is a really used feature, learn how to
		customize the image sizes from WordPress before activating a
		new theme.<br><br>
		Using the default sizes instead custom image sizes avoid
		longer media upload times and a smooth experience to your
		users.",
                }
              }
            />
            <p
              className="jsx-675934160 post-details"
            >
              Posted by 
              <span
                className="jsx-675934160"
              >
                @
                rafael angeline
              </span>
               
              in 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#cat"
                onClick={[Function]}
              >
                wordpress
              </a>
              , 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#dev"
                onClick={[Function]}
              >
                development
              </a>
            </p>
          </div>
          <div
            className="jsx-675934160 post"
          >
            <p
              className="jsx-675934160 date"
            >
              May 9, 2019
            </p>
            <h2
              className="jsx-675934160 title"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                Personal website overhaul.
              </a>
            </h2>
            <picture
              className="jsx-675934160"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                <img
                  alt="Alt Text"
                  className="jsx-675934160"
                  src="https://images.unsplash.com/photo-1508704019882-f9cf40e475b4?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=8c6e5e3aba713b17aa1fe71ab4f0ae5b&auto=format&fit=crop&w=800&q=60"
                  title="Image Title"
                />
              </a>
            </picture>
            <div
              className="jsx-675934160 excerpt"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "WordPress thumbnails is a really used feature, learn how to
		customize the image sizes from WordPress before activating a
		new theme.<br><br>
		Using the default sizes instead custom image sizes avoid
		longer media upload times and a smooth experience to your
		users.",
                }
              }
            />
            <p
              className="jsx-675934160 post-details"
            >
              Posted by 
              <span
                className="jsx-675934160"
              >
                @
                rafael angeline
              </span>
               
              in 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#cat"
                onClick={[Function]}
              >
                wordpress
              </a>
              , 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#dev"
                onClick={[Function]}
              >
                development
              </a>
            </p>
          </div>
          <div
            className="jsx-675934160 post"
          >
            <p
              className="jsx-675934160 date"
            >
              May 9, 2019
            </p>
            <h2
              className="jsx-675934160 title"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                Personal website overhaul.
              </a>
            </h2>
            <picture
              className="jsx-675934160"
            >
              <a
                className="jsx-675934160"
                href="http://localhost:3000/#"
                onClick={[Function]}
              >
                <img
                  alt="Alt Text"
                  className="jsx-675934160"
                  src="https://images.unsplash.com/photo-1508704019882-f9cf40e475b4?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=8c6e5e3aba713b17aa1fe71ab4f0ae5b&auto=format&fit=crop&w=800&q=60"
                  title="Image Title"
                />
              </a>
            </picture>
            <div
              className="jsx-675934160 excerpt"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "WordPress thumbnails is a really used feature, learn how to
		customize the image sizes from WordPress before activating a
		new theme.<br><br>
		Using the default sizes instead custom image sizes avoid
		longer media upload times and a smooth experience to your
		users.",
                }
              }
            />
            <p
              className="jsx-675934160 post-details"
            >
              Posted by 
              <span
                className="jsx-675934160"
              >
                @
                rafael angeline
              </span>
               
              in 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#cat"
                onClick={[Function]}
              >
                wordpress
              </a>
              , 
              <a
                className="jsx-2186686606"
                href="http://localhost:3000/#dev"
                onClick={[Function]}
              >
                development
              </a>
            </p>
          </div>
        </div>
        <nav
          className="jsx-662577867"
        >
          <div
            className="jsx-662577867 page-numbers"
          >
            <span
              className="jsx-662577867 current"
            >
              1
            </span>
            <a
              className="jsx-662577867"
              href="http://localhost:3000/undefined/2"
              onClick={[Function]}
            >
              2
            </a>
            <a
              className="jsx-662577867"
              href="http://localhost:3000/undefined/3"
              onClick={[Function]}
            >
              3
            </a>
            <span
              className="jsx-662577867 "
            >
              ...
            </span>
            <a
              className="jsx-662577867"
              href="http://localhost:3000/undefined/5"
              onClick={[Function]}
            >
              5
            </a>
          </div>
          <div
            className="jsx-662577867 page-arrows"
          >
            <a
              className="jsx-662577867 page-arrows-right"
              href="http://localhost:3000/undefined/2"
              onClick={[Function]}
            >
              <svg
                className="jsx-662577867 feather feather-chevron-right"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="1"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  className="jsx-662577867"
                  points="9 18 15 12 9 6"
                />
              </svg>
            </a>
          </div>
        </nav>
      </div>
      <aside
        className="jsx-1079858574"
      >
        <div
          className="jsx-1688206069 widget"
        >
          <h3
            className="jsx-1688206069"
          >
            Categories.
          </h3>
          <ul
            className="jsx-1688206069"
          >
            <li
              className="jsx-1688206069"
            >
              <a
                className="jsx-1688206069"
                href="http://localhost:3000/#1"
                onClick={[Function]}
              >
                Category 1
              </a>
            </li>
            <li
              className="jsx-1688206069"
            >
              <a
                className="jsx-1688206069"
                href="http://localhost:3000/#2"
                onClick={[Function]}
              >
                Category 2
              </a>
            </li>
            <li
              className="jsx-1688206069"
            >
              <a
                className="jsx-1688206069"
                href="http://localhost:3000/#3"
                onClick={[Function]}
              >
                Category 3
              </a>
            </li>
          </ul>
        </div>
      </aside>
    </div>
  </main>,
  <footer
    className="jsx-7675024"
  >
    <div
      className="jsx-7675024 container"
    >
      <div
        className="jsx-7675024 widget"
      >
        <h3
          className="jsx-7675024"
        >
          Contact.
        </h3>
        <p
          className="jsx-7675024"
        >
          Rafael Angeline
          <br
            className="jsx-7675024"
          />
          Send me your message
          <br
            className="jsx-7675024"
          />
          me [at] rafaelangeline.com
        </p>
      </div>
      <div
        className="jsx-7675024 widget"
      >
        <h3
          className="jsx-7675024"
        >
          Location.
        </h3>
        <p
          className="jsx-7675024"
        >
          Magic Happens at
          <br
            className="jsx-7675024"
          />
          São Paulo, SP - Brazil
        </p>
      </div>
      <div
        className="jsx-7675024 widget"
      >
        <h3
          className="jsx-7675024"
        >
          Stay in Touch.
        </h3>
        <p
          className="jsx-7675024"
        >
          Follow me on
           
          <a
            className="jsx-7675024 twitter"
            href="https://twitter.com/RafaelAngeline"
            target="_blank"
          >
            <span
              className="jsx-7675024"
            >
              Twitter
            </span>
          </a>
          ,
           
          <a
            className="jsx-7675024 stackoverflow"
            href="http://stackexchange.com/users/8383862/rafael-angeline"
            target="_blank"
          >
            <span
              className="jsx-7675024"
            >
              StackOverflow
            </span>
          </a>
          ,
           
          <a
            className="jsx-7675024 github"
            href="https://github.com/raphaelkross"
            target="_blank"
          >
            <span
              className="jsx-7675024"
            >
              Github
            </span>
          </a>
          ,
           
          <a
            className="jsx-7675024 wordpress"
            href="https://profiles.wordpress.org/rafaelangeline"
            target="_blank"
          >
            <span
              className="jsx-7675024"
            >
              WordPress
            </span>
          </a>
        </p>
      </div>
      <p
        className="jsx-7675024 copy"
      >
        © Copyright Rafael Angeline. Built by Rafael Angeline with React, Next.js and WordPress &lt;3
      </p>
    </div>
  </footer>,
]
`;
